EIPS/eip-6780.md

```diff
@@ -0,0 +1,65 @@
+---
+eip: 6780
+title: SELFDESTRUCT only in same transaction
+description: SELFDESTRUCT will recover all funds to the caller but not delete the account, except when called in the same transaction as creation
+author: Guillaume Ballet (@gballet), Vitalik Buterin (@vbuterin), Dankrad Feist (@dankrad)
+discussions-to: https://ethereum-magicians.org/t/deactivate-selfdestruct-except-where-it-occurs-in-the-same-transaction-in-which-a-contract-was-created/13539
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-03-25
+requires: 2681, 2929, 3529
+---
+
+## Abstract
+
+This EIP changes the functionality of the `SELFDESTRUCT` opcode. The new functionality will be only to send all Ether in the account to the caller, except that the current behaviour is preserved when `SELFDESTRUCT` is called in the same transaction a contract was created.
+
+## Motivation
+
+The `SELFDESTRUCT` opcode requires large changes to the state of an account, in particular removing all code and storage. This will not be possible in the future with Verkle trees: Each account will be stored in many different account keys, which will not be obviously connected to the root account.
+
+This EIP implements this change. Applications that only use `SELFDESTRUCT` to retrieve funds will still work. Applications that only use `SELFDESTRUCT` in the same transaction 
+
+## Specification
+
+The behaviour of `SELFDESTRUCT` is changed in the following way:
+
+1. Case where `SELFDESTRUCT` is executed in a transaction that is not the same as the contract calling `SELFDESTRUCT` was created:
+
+ - Does not delete any storage keys or code
+ - Transfer the account balance to the target **and** set account balance to `0.`
+ - Note that no refund is given since [EIP-3529](./eip-3529.md).
+ - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+  
+2. Case where `SELFDESTRUCT` is executed in the same transaction the contract was created
```

Comment:
```suggestion
2. When `SELFDESTRUCT` is executed in the same transaction as the contract was created:
```
Regardless of the first word change, please add the colon for consistency.

Alternatively (this is my preference), simply remove the second section.